EEJT-000A/main/java/task2/SearchingController.java

```diff
@@ -0,0 +1,46 @@
+package task2;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+public class SearchingController {
+    private Parameters parameters;
+    private SearchByParam searchByParam;
+
+    public SearchingController(Parameters parameters) {
+        this.parameters = parameters;
+    }
+
+    public List<String> doChain(List<String> list) {
+        if (!isNull(parameters)) {
+            if (isNull(list)) {
+                list = new ArrayList<>();
+            }
+            if ((parameters.getSizeMore()) != null
+                    && (parameters.getSizeLess()) != null) {
+                SearchByParam newSearchByParam = null;
+                searchByParam = new SearchBySizeChange(newSearchByParam);
+            }
```

Comment:
Optional task with *: Try to replace all those `if`s with a map for example.
Expected pseudocode structure:
```java
           for (Map.Entry<UserInputChecker, SearchByParamCreator> entry : yourMap.entrySet()) {
                if (entry.getKey().testUserInput(...)) {
                    searchByParam = entry.getValue().createCorrespondingSeachByParamClass(searchByParam));
                }
            }
```

Note/Hint: `UserInputChecker` and `SearchByParamCreator` just random class names I came up. You can use OOTB java functional interfaces instead like `Predicate<Parameters>` for user input check and `Funtion<SearchByParam, SearchByParam>` for corresponding chain part creation.